### ISSUE TITLE
SWIG/_m2crypto_wrap.c:32789:120: error: stray ‘\’ in program

### ISSUE THREAD

**Issue by jhihn** [swig/swig#1818]
This is on a Raspberry Pi 4 (32bit raspbian)

```
SWIG/_m2crypto_wrap.c:32789:120: error: stray ‘\’ in program 
SWIG_Python_SetConstant(d, d == md ? public_interface : NULL, "__WCHAR_MAX",SWIG_From_wchar_t((wchar_t)((0x7fffffff+L\'\\0\'))));
```

Clearly the `L\'` should only be `L'`

This is after I fixed a FPU issue:
Because for whatever reason it was thinking I didn't have a hardware FPU:
```
Running ak-M2Crypto-0.26.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-xX3yqD/ak-M2Crypto-0.26.1/egg-dist-tmp-4hjODh
/usr/include/arm-linux-gnueabihf/gnu/stubs.h:7: Error: Unable to find 'gnu/stubs-soft.h'
error: Setup script exited with error: command 'swig' failed with exit status 1
```

Required:
```
cd /usr/include/arm-linux-gnueabihf/gnu
ln -s stubs-hard.h stubs-soft.h
sudo ln -s stubs-hard.h stubs-soft.h
```

**Comment by wsfulton** [comment]
We need to see the input to help. What does the constant look like? You can use `swig -E` to look at the preprocessed output, which is what will be the actual input.

**Comment by jhihn** [comment]
The constant is just that `0x7fffffff+L\'\\0\'`
Apparently this al stems from invalid soft-fpu detection on RasberryPi which has a hardware FPU.

**Comment by ojwb** [comment]
@jhihn That's presumably the generated code - what does the constant being wrapped to give this look like in the **INPUT** that was fed into SWIG to generate that code?

E.g. this doesn't reproduce the problem:

```
%module test
%include <wchar.i>
%inline %{
# define __WCHAR_MAX	(0x7fffffff + L'\0')
%}
```

Probably you don't want to be wrapping this constant in fact (from the constant name it looks like you're asking SWIG to wrap everything in `/usr/include/<whatever the triplet is>/bits/wchar.h` for some reason) but it should still generate valid wrapper code.

> Apparently this al stems from invalid soft-fpu detection on RasberryPi which has a hardware FPU.

I can't see a way any of that could matter to SWIG.

Also, what SWIG version are you using?  We have fixed at least one bug which resulted in output something like yours, though I don't recall one recently.

**Comment by ojwb** [comment]
Aha, this was fixed in SWIG 4.0.2 by 90ba174fcea1618af57aa594199541d47a89b7f6.
